### DISCUSSION TITLE
Is it possible to put mpsc channel into a state

### DISCUSSION THREAD

**Discussion by oOBoomberOo** [gotham-rs/gotham#584] [Q&A] [answered]
I ran into an issue with storing Tokio's `mpsc::UnboundedSender` into the state middleware to send a message to another thread to do some task. Running the task directly from the request **isn't** an option since this is a long-running I/O task.

The issue seems to be that the channel's implementation used `UnsafeCell`, causing the error below.
```
error[E0277]: the type `UnsafeCell<AtomicUsize>` may contain interior mutability and a reference may not be safely transferrable across a catch_unwind boundary
```
Is this something doable within this framework, or is there any workaround for this?

**Comment by msrd0** [comment, marked as answer]
This looks like a similar problem to https://github.com/gotham-rs/gotham/issues/570 - it might not be possible to store that sender in the state if it isn't unwind safe.

I guess the work around would be to not store non-unwind safe structs in the state? You can always store them somewhere else (e.g. in a static variable) and use that from your handler - handler's are just Rust functions after all.

**Reply by oOBoomberOo** [reply]
Hmm, I think that should work for my use case.
